### PR TITLE
BrowserStack.java- Roobini

### PR DESCRIPTION
**Pull request by sroobini**
BrowserStack.java
y.ho
f.bk
y.be
g.co
y.ho
f.bk
y.be
g.co
y.be
Stack is empty
DONE

### CODE REVIEW

File: BrowserStack.java [hunk 1]
```diff
@@ -37,33 +45,96 @@ public static void myassert(boolean  x) {
...
         //WRITE YOUR CODE HERE
-
-    }
+    	String popped;
+    	if(top<0){
+    		popped = "Stack is empty";
+    	}
+    	else{
+    	popped = myStack[top];
+    	myStack[top] = null;
```
Comment (manojgo) [inline BrowserStack.java:63]: good. this takes care of marking the object as garbage immediately.

File: BrowserStack.java [hunk 2]
```diff
@@ -37,33 +45,96 @@ public static void myassert(boolean  x) {
...
+    	top--;
+    	return popped;
+    } 
 
     private void printStack(){
         //WRITE YOUR CODE HERE
-
-    }
+    	if(isEmpty()){
+    		System.out.println("Stack is empty");
```
Comment (manojgo) [inline BrowserStack.java:72]: we can return from here.

File: BrowserStack.java [hunk 3]
```diff
@@ -13,19 +15,25 @@
...
+    int STACK_SIZE = 2; //INITIAL STACK SIZE, GROW IF ITS NEEDED
 
     //Constructor to initialize your stack
     BrowserStack(){
         this.myStack = new String[STACK_SIZE];
         this.top = -1;
     }
+    private void createNewStack(){
+    	String[] newStack = new String[STACK_SIZE * 2];
+    	newStack = Arrays.copyOf(myStack, newStack.length);
```
Comment (manojgo) [inline BrowserStack.java:33]: Arrays.copyOf() method apart from copying data, also creates a new array for the copied elements. So, in the previous line the new String[] constructed is thrown away. If you don't want that to be thrown away, use Systems.arrayCopy().

File: BrowserStack.java [hunk 4]
```diff
@@ -13,19 +15,25 @@
...
+    	STACK_SIZE = STACK_SIZE *2;
+    	myStack = newStack;
```
Comment (manojgo) [inline BrowserStack.java:35]: Old myStack is thrown away and so all the old elements in it. In this program it is ok.

File: BrowserStack.java [hunk 5]
```diff
@@ -37,33 +45,96 @@ public static void myassert(boolean  x) {
...
+    	if(top==(STACK_SIZE-1)){
+    		return true;
+    	}
+    	return false;
     }
+    private  boolean search(String url){
+        //WRITE YOUR CODE HERE
+    	int temptop = -1;
+    	boolean ispresent =false;
+    	String[] tempStack = new String[myStack.length];
```
Comment (manojgo) [inline BrowserStack.java]: tempStack can be of length "top".

File: BrowserStack.java [hunk 6]
```diff
@@ -37,33 +45,96 @@ public static void myassert(boolean  x) {
...
 
     private void printStack(){
         //WRITE YOUR CODE HERE
-
-    }
+    	if(isEmpty()){
+    		System.out.println("Stack is empty");
+    	}
+    	int temptop = -1;
+    	String[] tempStack = new String[myStack.length];
```
Comment (manojgo) [inline BrowserStack.java:75]: tempStack can be of length "top".

File: BrowserStack.java [hunk 7]
```diff
@@ -37,33 +45,96 @@ public static void myassert(boolean  x) {
...
+    	while(temptop>=0){
+    		push(tempStack[temptop]);
+    		temptop--;
+    	}    	
+    	
+    } 
     private  String peek(){
         //WRITE YOUR CODE HERE
+         String peeked;
+    	peeked = myStack[top];
```
Comment (manojgo) [inline BrowserStack.java:91]: good to check if the top index is >= 0 here also for safety.

File: BrowserStack.java [hunk 8]
```diff
@@ -37,33 +45,96 @@ public static void myassert(boolean  x) {
...
+    		return true;
+    	}
+    	return false;
     }
+    private  boolean search(String url){
+        //WRITE YOUR CODE HERE
+    	int temptop = -1;
+    	boolean ispresent =false;
+    	String[] tempStack = new String[myStack.length];
+    	while(top>=0){
```
Comment (manojgo) [inline BrowserStack.java:121]: better to use !isEmpty() here.

**Review by manojgo — COMMENTED** [review]
Overall looks very good to me. Keep rocking.

File: BrowserStack.java (hunk 3 above)
Comment (sroobini) [reply BrowserStack.java:33]: Thanks for the update... very useful

**Pull request closed without merging**